Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -391,6 +391,12 @@ public override void Update(float frameTime)
                     _damageableSystem.TryChangeDamage(uid, flammable.Damage * damageScale);
 
                     AdjustFireStacks(uid, flammable.FirestackFade * (flammable.Resisting ? 10f : 1f), flammable);
+
+                    if (!flammable.OnFire)
+                    {
+                        _alertsSystem.ClearAlert(uid, AlertType.Fire);
+                        continue;
+                    }
```

Comment:
this should be in Extinguish() which is what's gonna get called by AdjustFireStacks if the ent is not longer on fire.

Reply:
Extinguish() already sets OnFire to false. I added this not to continue to set the `ignitionSource` to ignited if fire stacks were just adjusted to zero and the entity was thus extinguished. Can you elaborate what to do in extinguish instead to prevent that?

I also realised another thing. Maybe update should first check oxygen level and set ignition, then adjust fire stacks. This also solves the primary problem. I pushed this suggestion, let me know if it's better or not.